compare_platforms/compare_platforms_read_only.c

```diff
@@ -0,0 +1,189 @@
+{
+    printf("\nOverflowing reviewer realname by %zx\n", reps);
+    memset(rv->realname + offset, 'w', reps);
+    print_details(rv);
+}
+
+int main()
+{
+    struct review review;
+    // In a non-CHERI environment, write protection is weakly enforced
+    bool bWeak = true;
+    
+    size_t smallsz = 0x20;
+    size_t biggersz = 0x1001;
+    
+    char *username = malloc(smallsz);
+    strcpy(username, "Reviewer 2"); 
+    review.username = username;
+       
+    char *realname = malloc(smallsz);
+    strcpy(realname, "Baba Yaga");
+    review.realname = realname;
+    
+    char *privatereview = malloc(biggersz);
+    strcpy(privatereview, "I cannot believe I read this appalling piece of dreck from start to finish. The authors should be ashamed, and I hope I get an opportunity to tell them so to their faces.");
+    review.privatereview = privatereview;
+    
+    char *publicreview = malloc(biggersz);
+    strcpy(publicreview, "I am a little unclear as to the contribution. I think the authors could strengthen their case considerably if they conducted an RCT. Weak reject.");
+    review.publicreview = publicreview;
+    
+    char *permissions = malloc(smallsz);
+    strcpy(permissions, "r");
+    review.permissions = permissions;
+    
+    print_details(&review);
+    
+    assert((size_t)(username + smallsz) <= (size_t)realname); 
+    assert((size_t)(privatereview + biggersz) <= (size_t)publicreview);
+    assert((size_t)(realname + smallsz) <= (size_t)permissions);
+    
+    bool b_improved = false;
+    char *newpublicreview = malloc(biggersz);
+    strcpy(newpublicreview, "5 ***** strong accept. This author deserves two Nobels and an ice cream.\n");
+
+#ifdef __CHERI_PURE_CAPABILITY__
+    printf("\nsmallsz=%zx, CRRL(smallsz)=%zx\n", smallsz,
+        __builtin_cheri_round_representable_length(smallsz));
+    printf("biggersz=%zx, CRRL(biggersz)=%zx\n", biggersz,
+        __builtin_cheri_round_representable_length(biggersz));
+    printf("username=%#p realname=%#p diff=%tx\n", username, realname, realname - username);
+    printf("realname=%#p privatereview=%#p diff=%tx\n", realname, privatereview, privatereview - realname);
+    printf("privatereview=%#p publicreview=%#p diff=%tx\n", privatereview, publicreview, publicreview - privatereview);
+    printf("publicreview=%#p permissions=%#p diff=%tx\n", publicreview, permissions, permissions - publicreview);
+    printf("realname=%#p permissions=%#p diff=%tx\n", realname, permissions, permissions - realname);
+    
+    if(can_write(&review) == false) {
+        printf("\nSetting the review to read-only.\n");
+        struct review *ro_review = set_read_only(&review);
+        assert((cheri_perms_get(ro_review) & CHERI_PERM_STORE) == 0);
+
+        printf("\nThe struct is read-only so trying to change the review will make the program crash.\n");
+        printf("\nTry anyway? y/n: \n");
+        char answer;
+        scanf(" %c", &answer);
```

Comment:
Rather than requesting user input, can we make it a command-line flag?

Reply:
Fixed in 66e958a761fdeb42fa4e944393fd607161ddcedc